packages/pockethost.io/src/routes/signup/+page.svelte

```
@@ -9,19 +9,21 @@
   let isFormButtonDisabled: boolean = true;
   $: isFormButtonDisabled = email.length === 0 || password.length === 0;
 
-  const handleSubmit = async(e) => {
+  const handleSubmit = async(e:SubmitEvent) => {
     e.preventDefault();
 
     isFormButtonDisabled = true;
 
-    await handleRegistration(email, password, (error) => {
-      formError = error;
-    })
+    try {
+      await handleRegistration(email, password)
```

The functions in the `database.ts` file already have try/catch calls. Are the extra ones needed?

---

Yes because we want each step to throw so we can stop at the first error rather than continue.

I woke up this morning realizing I still didn't get it quite right. For example what if the account registration and login succeeds, but the instance name is taken and so that 3rd step fails. On the next submit, it needs to skip account creation and login and just go directly to instance creation. 

Also I changed the other methods so they only throw if the `setError` method is omitted.